src/query/service/src/sessions/session_privilege_mgr.rs

```diff
@@ -267,14 +268,20 @@ impl SessionPrivilegeManager for SessionPrivilegeManagerImpl {
 
     #[async_backtrace::framed]
     async fn has_ownership(&self, object: &OwnershipObject) -> Result<bool> {
-        let role_mgr = RoleCacheManager::instance();
+        let user_mgr = UserApiProvider::instance();
         let tenant = self.session_ctx.get_current_tenant();
-
-        // if the object is not owned by any role, then considered as ACCOUNT_ADMIN, the normal users
-        // can not access it unless ACCOUNT_ADMIN grant relevant privileges to them.
-        let owner_role_name = match role_mgr.find_object_owner(&tenant, object).await? {
-            Some(owner_role) => owner_role.name,
+        let owner_role_name = match user_mgr.get_ownership(&tenant, object).await? {
             None => BUILTIN_ROLE_ACCOUNT_ADMIN.to_string(),
+            Some(owner) => match user_mgr.get_role(&tenant, owner.role).await {
+                Ok(owner_role) => owner_role.name,
+                Err(e) => {
+                    if e.code() == ErrorCode::UNKNOWN_ROLE {
+                        BUILTIN_ROLE_ACCOUNT_ADMIN.to_string()
```

Comment:
IMHO this unknown role fallback logic could be reduced as the logic has already been encapsulated in the `get_ownership` call, so just keep the match statement as before is ok? 

```
let owner_role_name = match role_mgr.find_object_owner(&tenant, object).await? {
            Some(owner_role) => owner_role.name,
            None => BUILTIN_ROLE_ACCOUNT_ADMIN.to_string(),
}
```

Reply:
find_object_owner is use rolecache. So maybe we can replace it with get real meta ownership into.

Reply:
find_object_owner is ok. Now load from cache do not have bug.

Reply:
oh, sorry my comment made a typo in the function name:

IMO `role_mgr.get_ownership` is better, the cache do not offer many benefits here.

but as we have already encapsulated the logic about fallback to BUILTIN_ROLE_ACCOUNT_ADMIN, we can simplify the code as the following, do not need handle the UNKNOWN_ROLE error here again:

```
let owner_role_name = match role_mgr.get_ownership(&tenant, object).await? {
            Some(owner_role) => owner_role.name,
            None => BUILTIN_ROLE_ACCOUNT_ADMIN.to_string(),
}
```